mappings/net/minecraft/structure/rule/RuleTest.mapping

```diff
@@ -1,5 +1,6 @@
 CLASS net/minecraft/class_3825 net/minecraft/structure/rule/RuleTest
 	COMMENT Rule tests are used in structure generation to check if a block state matches some condition.
+	FIELD field_25012 TYPE_CODEC Lcom/mojang/serialization/Codec;
```

Comment:
Is this actually like `Registry.RULE_TEST.dispatch(xx)`? This is a codec for a whole rule test, as that dispatch assigns the type's codec as decoder in decoding branch (i.e. read "type" field for the type, and get the right decoder for the contents), and has a method ref to `getType` to obtain the type's codec for encoding on the other hand.

Just curious, what is the origin of the `TYPE_CODEC` names? Meaning it's a codec for the type only?

Reply:
Yeah, that is usually what it means. I've also co-opted it to mean the codec for the base type, so conflicts with subclasses are avoided

Reply:
should we call it BASE_CODEC then?

Reply:
Probably a good idea, yeah. We should put codec guidelines into the javadoc conventions I think, to standardize this

Reply:
Guess we will do that in a later pr. This looks good enough.